atlas/core/imageManager.cpp

```diff
@@ -48,94 +48,95 @@ namespace imageManager
+				|| ( temp_image.height() > webp_max )		// Is it too tall for WebP?
+		    )
 			{
-				const auto hash { hashData( byteArray, static_cast< int >( byteArray.size() ) ) };
+				//Buffer is larger. Meaning webp is worse. Don't use it.
+				const auto hash { hashData( byteArray.data(), static_cast< int >( byteArray.size() ) ) };
 				const auto dest { dest_root / ( hash.toHex().toStdString() + path.extension().string() ) };
 
-				//Qt is stupid and will not save gifs...  so we have to copy it
-				const bool file_copied { std::filesystem::copy_file( path, dest ) };
-				if ( !file_copied )
-					throw std::
-						runtime_error( fmt::format( "Unable to save gif to images folder: {}", path.filename() ) );
+				const QImage img { QImage::fromData( byteArray ) };
+				img.save( QString::fromStdString( dest.string() ) );
 
 				return dest;
 			}
 			else
 			{
-				const std::string image_type { config::images::image_type::get().toStdString() };
-
-				TracyCZoneN( tracy_SaveImage, "Image save to buffer as WEBP", true );
-				QByteArray webp_byteArray;
-				QBuffer webp_buffer( &webp_byteArray );
-				temp_image.save( &webp_buffer, image_type.c_str(), 90 );
-				TracyCZoneEnd( tracy_SaveImage );
-
-				//Which is smaller?
-				if ( webp_buffer.size() < byteArray.size() )
-				{
-					//Buffer is smaller. Meaning webp is smaller. Use it
-					const auto hash { hashData( webp_buffer.data(), static_cast< int >( webp_buffer.size() ) ) };
-					const auto dest { dest_root / ( hash.toHex().toStdString() + ".webp" ) };
-
-					// We shouldn't need to make a new image since we already have it? (tmp_image)
-					// const QImage img { QImage::fromData( webp_byteArray ) };
-					temp_image.save( QString::fromStdString( dest.string() ) );
-
-					return dest;
-				}
-				else
-				{
-					//Buffer is larger. Meaning webp is worse. Don't use it.
-					const auto hash { hashData( byteArray.data(), static_cast< int >( byteArray.size() ) ) };
-					const auto dest { dest_root / ( hash.toHex().toStdString() + path.extension().string() ) };
-
-					const QImage img { QImage::fromData( byteArray ) };
-					img.save( QString::fromStdString( dest.string() ) );
-
-					return dest;
-				}
-			}
-		}
+				//Buffer is smaller. Meaning webp is smaller. Use it
+				const auto hash { hashData( webp_buffer.data(), static_cast< int >( webp_buffer.size() ) ) };
+				const auto dest { dest_root / ( hash.toHex().toStdString() + ".webp" ) };
 
-		else
-		{
-			atlas::logging::userwarn( fmt::format(
-				"importImage failed. Attempted to open file {} which doesn't exist anymore. Wrong permissions?",
-				path ) );
-			throw std::runtime_error( fmt::format( "Filepath {} does not exist. Unable to add as image", path ) );
+				// We shouldn't need to make a new image since we already have it? (tmp_image)
+				// const QImage img { QImage::fromData( webp_byteArray ) };
+				temp_image.save( QString::fromStdString( dest.string() ) );
```

Comment:
This can be cleaned up since the comment was for tower originally. Ideally we should just write `webp_byteArray` to disk. Unsure exactly why this was being done this way to begin with. Ask @towerwatchman if maybe there is a reason for it?